WordPress/openverse-api
# Add validate spec command

## Description


Adds a command that we could run in CI to validate the schema. This is just a rough proposal. I see that it is not catching the lack of trailing slash on the `healthcheck` endpoint for example. Maybe there's some configuration we need to do with the `validate_spec`, like maybe pass a specific spec_url to use.

## Testing Instructions

Run `just build` so that you get the new dependencies. Then run `just dj validateopenapischema` and verify that it does not fail with validation errors.

If you want to see it working (i.e., failing) you can undo some of the changes I made like removing the trailing slashes from some URL configurations or adding back that `unique=True` on the auth_token endpoint configuration. The latter yields a very mysterious error that took me quite a while to find a solution for! This webapp gave me a clear, descriptive error that was much more helpful than the validate_spec error: https://editor.swagger.io/

Maybe there's a way to use the validation underlying that in the future as an iterative improvement on this.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
